src/codegen/compile.ml

```diff
@@ -10630,21 +10642,23 @@ and compile_prim_invocation (env : E.t) ae p es at =
     SR.Vanilla,
     compile_array_index env ae e1 e2 ^^
     load_ptr
-  | NextArrayOffset spacing, [e] ->
```

Comment:
I guess I had `spacing` as a parameter to allow `Nat8` (relevant for `Blob`?) and `Float` (but these are currently heap-indirect) increments. We can recall this feature if needed in the future.

Reply:
I think spacing would increment by 4 or 1 depending on vals or keys iteration. But I can't store a byte offset with only 30 29 bits available, so just shift by 2 in both cases now.